RecoLocalCalo/HcalRecAlgos/interface/HcalDeterministicFit.h

```diff
@@ -62,4 +65,109 @@ class HcalDeterministicFit {
   
 };
 
+template<class Digi>
+void HcalDeterministicFit::apply(const CaloSamples & cs, const std::vector<int> & capidvec, const HcalCalibrations & calibs, const Digi & digi, std::vector<double> & HLTOutput) const {
+  std::vector<double> corrCharge;
+  std::vector<double> inputCharge;
+  std::vector<double> inputPedestal;
+  const unsigned int cssize = cs.size();
+  double gainCorr = 0;
+  for(unsigned int ip=0; ip<cssize; ++ip){
+    if( ip >= (unsigned) 10 ) continue; // Too many samples than what we wanna fit (10 is enough...) -> skip them
```

Comment:
@kfiekas 
`int const maxIteration = 10;`
is much more clear.
Anyhow, not clear to me why 10 is enough... Maybe this should become a parameter in the local reco?
@slava77 I let you comment more: I am definitively not an expert on this...